pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -104,3 +105,63 @@ def merge_sort_parallel(array, num_threads, **kwargs):
 
     if _check_type(array, DynamicArray):
         array._modify(force=True)
+
+def brick_sort(array, **kwargs):
+    """
+    Implements Brick Sort / Odd Even sorting algorithm
+
+    Parameters
+    ==========
+
+    array: Array
+        The array which is to be sorted.
+    start: int
+        The starting index of the portion
+        which is to be sorted.
+        Optional, by default 0
+    end: int
+        The ending index of the portion which
+        is to be sorted.
+        Optional, by default the index
+        of the last position filled.
+    comp: lambda/function
+        The comparator which is to be used
+        for sorting. If the function returns
+        False then only swapping is performed.
+        Optional, by default, less than or
+        equal to is used for comparing two
+        values.
+
+    Examples
+    ========
+    >>> from pydatastructs import OneDimensionalArray, brick_sort
+    >>> arr = OneDimensionalArray(int,[3, 2, 1])
+    >>> brick_sort(arr)
+    >>> [arr[0], arr[1], arr[2]]
+    [1, 2, 3]
+    >>> brick_sort(arr, comp=lambda u, v: u > v)
+    >>> [arr[0], arr[1], arr[2]]
+    [3, 2, 1]
+
+    References
+    ==========
+    .. [1] https://www.geeksforgeeks.org/odd-even-sort-brick-sort/
+    """
+    start = kwargs.get('start', 0)
+    end = kwargs.get('end', len(array) - 1)
+    comp = kwargs.get("comp", lambda u, v: u <= v)
+
+    is_sorted = False
+    while is_sorted is False:
+        is_sorted = True
+        for i in range(start+1, end, 2):
+            if _comp(array[i+1], array[i], comp):
+                array[i], array[i+1] = array[i+1], array[i]
+                is_sorted = False
+        for i in range(start, end, 2):
+            if _comp(array[i+1], array[i], comp):
+                array[i], array[i+1] = array[i+1], array[i]
+                is_sorted = False
```

Comment:
It says _comp not found.
`Fail: NameError: name '_comp' is not defined`

Reply:
merge master to your branch `bricksort`, push the changes here.

Reply:
I think definition for _comp was accidentally removed @czgdp1807

Reply:
Thanks it worked